File: connectors/connector-mantis/src/main/java/com/taskadapter/connector/mantis/MantisManagerFactory.java
package com.taskadapter.connector.mantis;

import com.taskadapter.connector.definition.WebConnectorSetup;

public class MantisManagerFactory {
    public static MantisManager createMantisManager(WebConnectorSetup setup) {
        if (setup.isUseApiKey()) {
            throw new RuntimeException("authorization using API key is not supported for Mantis Connector");
        }
        return new MantisManager(setup.getHost(), setup.getUserName(), setup.getPassword());
    }
}

File: connectors/connector-mantis/src/main/scala/com/taskadapter/connector/mantis/MantisManagerFactory.scala
package com.taskadapter.connector.mantis

import com.taskadapter.connector.definition.WebConnectorSetup

object MantisManagerFactory {
  def createMantisManager(setup: WebConnectorSetup): MantisManager = {
    if (setup.useApiKey) {
      throw new RuntimeException("authorization using API key is not supported for Mantis Connector")
    } else {
      new MantisManager(setup.host, setup.userName, setup.password)
    }
  }
}
